jlm/llvm/opt/alias-analyses/PointerObjectSet.hpp

```diff
@@ -69,47 +66,95 @@ class PointerObject final
       MarkAsEscaped();
   }
 
-  PointerObjectKind
+  [[nodiscard]] PointerObjectKind
   GetKind() const noexcept
   {
     return Kind_;
   }
 
+  /**
+   * Some PointerObjects are not capable of pointing to anything else.
+   * Their points-to-set will always be empty, and constraints that attempt
+   * to add pointees should be no-ops that are silently ignored.
+   * The same applies to attempts at setting the PointsToExternal-flag.
+   * @return true if this PointerObject can point to other PointerObjects
+   */
+  [[nodiscard]] bool
+  CanPoint() const noexcept
+  {
+    return Kind_ != PointerObjectKind::FunctionMemoryObject;
+  }
+
+  /**
+   * Some PointerObjects don't have addresses, and can as such not be pointed to.
+   * Any attempt at adding them to a points-to-set is a fatal error.
+   * @return true if this PointerObject can be pointed to by another PointerObject
+   */
+  [[nodiscard]] bool
+  CanBePointee() const noexcept
+  {
+    return Kind_ != PointerObjectKind::Register;
+  }
+
+  /**
+   * When the PointsToExternal-flag is set, the PointerObject possibly points to a storage
+   * instance declared outside to module, or to a memory object from this same module, that has
+   * escaped.
+   * @return true if the PointsToExternal is set.
+   */
   bool
   PointsToExternal() const noexcept
   {
     return PointsToExternal_;
   }
 
   /**
-   * Sets the PointsToExternal-flag.
+   * Sets the PointsToExternal-flag, if possible.
```

Comment:
It is always possible, but sometimes not necessary.

Reply:
I'm afraid it's no longer that simple. In an effort to avoid lambda nodes pointing to things, I decided to prevent PointerObjects of FunctionMemoryObject kind from ever pointing. I updated the comment to make this clearer.